src/lucky/renderable.cr

```diff
@@ -103,19 +103,66 @@ module Lucky::Renderable
     response.print
   end
 
+  private def handle_response(_response : Nil) forall T
+    # This is a hack to force Crystal to regenerate this method for
+    # each action. That way @type refers to the action with the problem,
+    # and not the first action to be compiled.
+    {% @type %}
```

Comment:
Thanks @asterite for the tip! And thanks @jwoertink for digging into this

Reply:
This is not needed because type is mentioned a bit below.

I think type need to be mentioned in the method that calls this method.

Reply:
in perform_action

Reply:
🤦‍♂ I happened to check this with an action that worked with the broken version. Thanks @asterite. Will give that a shot

Reply:
Hmm this caused a compiler bug:

```crystal
BUG: trying to downcast SignUps::New <- BrowserAction (Exception)
web    |   from Crystal::CodeGenVisitor#downcast_distinct<LLVM::Value, Crystal::Type+, Crystal::Type+>:NoReturn
```

But I think I had an idea that might work...we'll see :P